pkg/scheduler/scheduler_test.go

```diff
@@ -72,3 +79,253 @@ func TestSlaveStorage_getSlaveIds(t *testing.T) {
+	offers2 := []*mesos.Offer{offer2}
+	testScheduler.ResourceOffers(nil, offers2)
+
+	//add another offer from different slaveID
+	hostname2 := "h2"
+	offer3 := &mesos.Offer{Id: util.NewOfferID("test3"), Hostname: &hostname2, SlaveId: util.NewSlaveID(hostname2)}
+	offers3 := []*mesos.Offer{offer3}
+	testScheduler.ResourceOffers(nil, offers3)
+
+	//test precondition
+	assert.Equal(3, getNumberOffers(testScheduler.offers))
+	assert.Equal(2, len(testScheduler.slaves.getSlaveIds()))
+
+	//remove first slave
+	testScheduler.SlaveLost(nil, util.NewSlaveID(hostname))
+
+	//offers should be removed
+	assert.Equal(1, getNumberOffers(testScheduler.offers))
+	//slave hostnames should still be all present
+	assert.Equal(2, len(testScheduler.slaves.getSlaveIds()))
+
+	//remove second slave
+	testScheduler.SlaveLost(nil, util.NewSlaveID(hostname2))
+
+	//offers should be removed
+	assert.Equal(0, getNumberOffers(testScheduler.offers))
+	//slave hostnames should still be all present
+	assert.Equal(2, len(testScheduler.slaves.getSlaveIds()))
+
+	//try to remove non existing slave
+	testScheduler.SlaveLost(nil, util.NewSlaveID("notExist"))
+
+}
+
+//test when we loose connection to master we invalidate all cached offers
+func TestDisconnect(t *testing.T) {
+	assert := assert.New(t)
+
+	//
+	testScheduler := &KubernetesScheduler{
+		offers: offers.CreateRegistry(offers.RegistryConfig{
+			Compat: func(o *mesos.Offer) bool {
+				return true
+			},
+			// remember expired offers so that we can tell if a previously scheduler offer relies on one
+			LingerTTL:     defaultOfferLingerTTL * time.Second,
+			TTL:           defaultOfferTTL * time.Second,
+			ListenerDelay: defaultListenerDelay * time.Second,
+		}),
+		slaves: newSlaveStorage(),
+	}
+
+	hostname := "h1"
+	offer1 := &mesos.Offer{Id: util.NewOfferID("test1"), Hostname: &hostname, SlaveId: util.NewSlaveID(hostname)}
+	offers1 := []*mesos.Offer{offer1}
+	testScheduler.ResourceOffers(nil, offers1)
+	offer2 := &mesos.Offer{Id: util.NewOfferID("test2"), Hostname: &hostname, SlaveId: util.NewSlaveID(hostname)}
+	offers2 := []*mesos.Offer{offer2}
+	testScheduler.ResourceOffers(nil, offers2)
+
+	//add another offer from different slaveID
+	hostname2 := "h2"
+	offer3 := &mesos.Offer{Id: util.NewOfferID("test2"), Hostname: &hostname2, SlaveId: util.NewSlaveID(hostname2)}
+	offers3 := []*mesos.Offer{offer3}
+	testScheduler.ResourceOffers(nil, offers3)
+
+	//disconnect
+	testScheduler.Disconnected(nil)
+
+	//all offers should be removed
+	assert.Equal(0, getNumberOffers(testScheduler.offers))
+	//slave hostnames should still be all present
+	assert.Equal(2, len(testScheduler.slaves.getSlaveIds()))
+}
+
+//test we can handle different status updates, TODO check state transitions
+func TestStatus_Update(t *testing.T) {
+
+	mockdriver := MockSchedulerDriver{}
+	// setup expectations
+	mockdriver.On("KillTask", util.NewTaskID("test-task-001")).Return(mesos.Status_DRIVER_RUNNING, nil)
```

Comment:
probably want to execute `mockdriver.AssertExpectations(t)` to confirm that this was invoked